src/core_atmosphere/physics/physics_wrf/module_cu_gf_deep.F

```diff
@@ -2304,23 +2306,44 @@ subroutine cu_gf_deep_run(        &
           endif
       enddo
       !-----downdraft_cold_pool---------------
-      if (icoldpool > 0 .and. imid ==0) then
+      !if (icoldpool > 0 .and. imid ==0) then
+      if (icoldpool /= 0 .and. icoldpool /= 2 .and. imid ==0) then
       ! --- adding the gust front horizontal speed to the 2-d MCL wind
       ! --- only magnitude is augmented, direction is kept the same
       do i=its,itf
+          umcl(i)=0.
+          vmcl(i)=0.
+          dpsum=0.
           if(ierr(i) > 0 ) cycle
+           do k=kts+1,ktop(i)-1
+             trash =-(po_cup(i,k)-po_cup(i,kts))
+             if(trash.gt.300. .and. trash.lt.600.)then
+             dp=100.*(po_cup(i,k)-po_cup(i,k+1))
```

Comment:
indent for this if block

Reply:
The if block is used to only apply the convective clouds between 300hPa and 600hPa above surface.

Reply:
my suggestion is to indent these lines in the if block

Reply:
Updated following your suggestions. Thanks.